Processes.md

```diff
@@ -21,42 +21,114 @@ Up to date
 Maintain state.
 
 
-Let’s create a module Menuprocess.ex and see how we shall execute the process from the client in the menuserver.
+Let’s create a module MenuProcess.ex and see how we shall execute the process from the client in the menuserver.
 Below is a breakdown of a couple of processes in our module. Our module will enable three processes;
 Return the current menu state
 Add items to the menu
 Handle wrong entries
 
-For the client requests to be handled on the server, start_link is called, the function menu_state is spawned to its own
-process then passed the initial state which in our case will change from an empty [] to an actual list of items to be 
-added to the menu. The function responsible for that is shown below;
+start_link begins a spawn linked to the current process with the given functions, the pid is registered using `Process.register`. For the client requests to be handled on the server, start_link is called, the function menu_state is spawned to its own process then passed the initial state which in our case will change from an empty [] to an actual list of items to be added to the menu. The function responsible for that is shown below;
```

Comment:
This part is  a little ambiguous, it's hard to tell if this is true for all processes or just this one in particular